doozer/doozerlib/cli/config_tag_rpms.py

```diff
@@ -150,9 +151,34 @@ async def run(self):
                         logger.warning("Build %s is already tagged into %s", build["nvr"], entry.target_tag)
                         break
                     logger.info("Build %s is acceptable for %s", build["nvr"], entry.target_tag)
-                    builds_to_tag[entry.target_tag].add(build["nvr"])
+                    builds_to_tag[entry.target_tag][build["nvr"]] = build
                     break
 
+            if not entry.enforce_same_version:
+                continue
+
+            for tag, nvr_dict in builds_to_tag.items():
+                if not nvr_dict:
+                    continue
+
+                package_names = {b['name'] for b in nvr_dict.values()}
+                if package_names != {'kernel', 'kernel-rt'}:
+                    raise ValueError(f"Unexpected packages found with enforce_same_version: {package_names}")
+                if len(nvr_dict) != 2:
+                    raise ValueError(f"More than 2 kernel builds found {nvr_dict.keys()}")
+
+                kernel_build = next(b for b in nvr_dict.values() if b['name'] == 'kernel')
+                kernel_rt_build = next(b for b in nvr_dict.values() if b['name'] == 'kernel-rt')
+
+                # e.g. kernel-5.14.0-284.28.1.el9_2, kernel-rt-5.14.0-284.28.1.rt14.313.el9_2
+                kernel_version = f"{kernel_build['version']}-{split_el_suffix_in_release(kernel_build['release'])[0]}"
+                kernel_rt_version = f"{kernel_rt_build['version']}-{kernel_rt_build['release']}"
+                if kernel_version not in kernel_rt_version:
```

Comment:
The release parts of kernel and kernel-rt are not the same:
kernel: 284.28.1.el9_2
kernel-rt: 284.28.1.rt14.313.el9_2

This is what we do in gen-payload: https://github.com/openshift-eng/doozer/blob/15a672d3dd127feefaa99bbb293b9d20ee53fda1/doozerlib/cli/release_gen_payload.py#L1348

Reply:
Right I'm not matching them exactly but getting the non-el part of the release (split_el_suffix_in_release) from kernel "284.28.1" and checking if that exists somewhere in kernel_rt release string "284.28.1.rt14.313.el9_2" which works in this case. The gen-payload logic seems more brittle than this? but don't have a strong opinion, I can use that logic if you still think that's better

Reply:
Updated tests and code

Reply:
Ah, you made a good point. We don't have to check if their version-releases match or not.